regionServiceClient/usr/sbin/registercloudguest

```diff
@@ -193,45 +187,87 @@ if proxy:
                'https_proxy': https_proxy}
     logging.info('Using proxy settings: %s' % proxies)
 
-# Get the SMT server information from one of the configured region servers
+# Get the SMT server information from the metadats server or
+# one of the configured region servers
 response = None
-# Get the location of the cert files for the region servers
-cert_dir = cfg.get('server', 'certLocation')
-region_servers = cfg.get('server', 'regionsrv').split(',')
-random.shuffle(region_servers)
-for srv in region_servers:
-    srvName = srv.strip()
-    logging.info('Using region server: %s' % srvName)
-    certFile = cert_dir + '/' + srvName + '.pem'
-    if not os.path.isfile(certFile):
-        logging.info('No cert found: %s skip this server' % certFile)
-        continue
+try:
+    metadata_url = cfg.get('server', 'use_metadata_server')
+    logging.info(
+        'Using metada server "%s" to obtain SMT information' % metadata_url
+    )
     try:
-        response = requests.get('https://%s/%s' % (srvName, api),
-                                verify=certFile,
-                                timeout=15.0,
-                                proxies=proxies)
-        if response.status_code == 200:
-            break
-        else:
-            logging.error('=' * 20)
-            logging.error('Server returned: %d' % response.status_code)
-            logging.error(response.text)
-            logging.error('=' * 20)
-    except:
-        logging.error('No response from: %s' % srvName)
-        if srv == region_servers[-1]:
-            logging.error('None of the servers responded')
-            logging.error('\tAttempted: %s' % region_servers)
-            logging.error('Exiting without registration')
+        response = requests.get(
+            metadata_url,
+            timeout=15.0,
+            proxies=proxies
+        )
+        if response.status_code != 200:
+            raise Exception(
+                'Metadata server returned %s' % response.status_code
+            )
+    except Exception as e:
+        logging.error('=' * 20)
+        logging.error(e.message)
+        logging.error('Unable to obtain SMT server information, exiting')
+        sys.exit(1)
+    smt_info = json.loads(response.text)
+    expected_entries = ('fingerprint', 'SMTserverIP', 'SMTserverName')
+    smt_info_xml = '<regionSMTdata><smtInfo '
```

Comment:
Why not use something like [SimpleXMLWriter](http://effbot.org/zone/xml-writer.htm) ? Manually constructing XML as a string is pretty error-prone.